mm2src/coins/solana.rs

```diff
@@ -226,35 +230,32 @@ pub struct SolanaFeeDetails {
 }
 
 async fn withdraw_base_coin_impl(coin: SolanaCoin, req: WithdrawRequest) -> WithdrawResult {
-    let (to_send, my_balance) = coin.check_sufficient_balance(req.max, req.amount).await?;
-    let base_balance = coin.base_coin_balance().compat().await?;
-    let hash = coin.rpc().get_latest_blockhash().await?;
+    let (hash, fees) = coin.estimate_withdraw_fees().await?;
+    let (to_send, my_balance, sol_required) = coin.check_sufficient_balance(req.max, req.amount.clone(), fees).await?;
+    let lamports_to_send = if req.max {
+        sol_to_lamports(&my_balance)? - sol_to_lamports(&sol_required)?
```

Comment:
Does `estimate_withdraw_fees` estimates the fees correct enough?
What do you think about increasing the required fees by 5% like we do for ETH, ERC20, QRC20, UTXO coins:
https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/coins/eth.rs#L2811

It seems that there is no gas price in Solana, so maybe it's worth to increase `sol_required` by 5%?
It's not a blocker, so could you please learn about estimation in the next iteration?

Reply:
> 

I think this will be usefull when we implement swaps, for `withdraw` and this approach is almost instant so there is no possibility for the fees two changes in this timeframe. using `get_tx_fee_for_message` seems to be the most appropriate way to retrieve exact amount of fees to be paid atm